seidr/dev.py

```diff
@@ -71,26 +71,18 @@ def lexicase_ranking(candidates):
 
 def beam_search(beam, update, ranking=standard_ranking, beam_width=100):
     """Generic evolutionary algorithm for improving anything"""
-    new_beam = []
-
-    # yield beam_width draft (non-repaired) programs
-    for candidate in beam:
-        yield candidate
-        new_beam.append(candidate)
-
     while True:
-        beam = itertools.islice(ranking(new_beam), beam_width)
-        new_beam = []
-
-        # yield beam_width * branching_factor children (repaired programs)
-        for parent in beam:
-            for child in update(parent):
-                yield child
-                new_beam.append(child)
+        parents = []
+        for code in beam:
+            yield code
+            parents.append(code)
 
-        if len(new_beam) == 0:
+        if len(parents) == 0:
             break
 
+        parents = itertools.islice(ranking(parents), beam_width)
```

Comment:
When ranking is `lexicase_selection`, do we get `beam_width` programs in parents or just one?